src/box/lua/schema.lua

```diff
@@ -460,13 +476,37 @@ box.savepoint = function()
     return { csavepoint=csavepoint, txn_id=builtin.box_txn_id() }
 end
 
+local commit_options = {
+    is_sync = function(is_sync)
+        if type(is_sync) ~= "boolean" then
+            box.error(box.error.ILLEGAL_PARAMS,
+                      "is_sync must be a boolean")
+        end
+        return true
+    end,
+}
+
+box.commit = function(options)
+    local is_sync
+    if options then
+        check_param_table(options, commit_options)
+        is_sync = options.is_sync
+    end
+    if is_sync then
+        assert(builtin.box_txn_set_sync(is_sync) == 0)
+    end
+    if builtin.box_txn_commit() == -1 then
+        box.error()
+    end
```

Comment:
You can't call box_txn_commit via FFI since it yields

Read about details here https://github.com/tarantool/tarantool/commit/34234427a3cc1f181ef6583c36ed7e4193bafde7

Reply:
Well, this is not quite true.

The main problem of calling yielding function via FFI machinery is "conditional" yield (one can find an example in [test/app-luatest/lib/libyield.c](https://github.com/tarantool/tarantool/blob/master/test/app-luatest/lib/libyield.c)). Hence the trace can be compiled with no yield on in scope of the function called via FFI, but when yield happens, there is no other way, than call panic.

However, there is no problems with calling a function that yields unconditionally (i.e. `fiber_yield` is called each time the caller is called via FFI), but in that case it has less sense, since trace compilation is aborted and all FFI-related "JIT benefits" are gone.

I glanced on the patch and see no reason moving existing interfaces to FFI (I also found neither benchmarks, nor rationale in commit message/comment for such change).

Considering everything written above, I suggest to not use FFI for you changes, or provide a strong reasoning for this.

Reply:
afaik, box_txn_commit() will not yield if wal_mode is none. so this execution flow is conditionally yielding

Reply:
Done